eclipse-collections-code-generator/src/test/java/org/eclipse/collections/codegenerator/EclipseCollectionsCodeGeneratorTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2022 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.codegenerator;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Collection;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ParseResult;
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.ast.CompilationUnit;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class EclipseCollectionsCodeGeneratorTest
+{
+    @Test
+    public void validateApiTemplates() throws Exception
+    {
+        validateTemplates("api", 1338);
+    }
+
+    @Test
+    public void validateImplTemplates() throws Exception
+    {
+        validateTemplates("impl", 1469);
+    }
+
+    @Test
+    public void validateTestTemplates() throws Exception
+    {
+        validateTemplates("test", 2342);
+    }
```

Comment:
I'd prefer not to assert an exact expected number of files, or people will be forced to change it all the time. Asserting it's not an empty collection would be fine IMO.